be1-go/hub/standard_hub/hub_state/Peers.go

```diff
@@ -26,10 +27,17 @@ func NewPeers() Peers {
 }
 
 // AddPeerInfo adds a peer's info to the table
-func (p *Peers) AddPeerInfo(socketId string, info method.ServerInfo) {
+func (p *Peers) AddPeerInfo(socketId string, info method.ServerInfo) error {
 	p.Lock()
 	defer p.Unlock()
+
+	_, ok := p.peersInfo[socketId]
+	if ok {
+		return xerrors.Errorf("peersInfo already contains socketId %s for server %s", socketId, info)
+	}
```

Comment:
You should change `info` inside the `xerrors.Errorf` by the value you get from `p.peersInfo[socketId]` because we want to say that the socket is already used by an other server.

Reply:
Or something like 'peersInfo cannot add peer %s because socketId %s already exists'

Reply:
Do you thing that including both values of info could be useful ?
for example "peersInfo cannot add peer *new_value* because  it already contains peer *current_value* for socketID *id*"

Reply:
Yes but in this case you should only print the peerId without the servers informations for both

Reply:
peersInfo cannot add new peer new_peerID because socket socketID is already used by peer current_peerID

Reply:
peersInfo is a map of socketId to ServerInfo, maybe the following could be more readable ?
"cannot add *new_peer_info* because peersInfo[*id*] already contains *current_peer_info*"

Reply:
I don't think that adding the peer information other than its public key is useful in this case.